### DISCUSSION TITLE
useQuery argument runs multiple times even when Query is skipped

### DISCUSSION THREAD

**Discussion by AsuraKev** [reduxjs/redux-toolkit#4407] [Q&A] [answered]
Hi there,

I have the following query where the argument is a function that computes the data which the query needs. As you see, I dont want to run the Query until the conditions have been met, however I noticed that the buildSearchPayload was run multiple times.

Any idea?

thanks :)

  const {
    isFetching,
    isUninitialized,
    data: paginationResult,
    error: searchError,
  } = useSearchInvitationsQuery(buildSearchPayload(), { skip: fetchingDealers || fetchingMaskConfigurations});

**Comment by markerikson** [comment, marked as answer]
Standard React rules apply here - you probably want to put that `buildSearchPayload()` inside of a `useMemo()` hook.

**Reply by AsuraKev** [reply]
Thanks that worked. Waiting for React Compiler to address this!
